e2e/tests/test_cases/node_disconnection_test.robot

```diff
@@ -0,0 +1,95 @@
+*** Settings ***
+Documentation    Node disconnection test
+...              https://longhorn.github.io/longhorn-tests/manual/pre-release/node-not-ready/node-disconnection/node-disconnection/
+
+Test Tags    manual_test_case
+
+Resource    ../keywords/common.resource
+Resource    ../keywords/volume.resource
+Resource    ../keywords/setting.resource
+Resource    ../keywords/network.resource
+Resource    ../keywords/statefulset.resource
+Resource    ../keywords/workload.resource
+
+Test Setup    Set test environment
+Test Teardown    Cleanup test resources
+
+*** Variables ***
+${LOOP_COUNT}    3
+${RETRY_COUNT}    300
+${RETRY_INTERVAL}    1
+
+*** Test Cases ***
+Node Disconnect And Keep Data Writing And No Replica On The Disconnected Node
+    Given Set setting auto-salvage to false
+
+    When Create volume 0 with 10 GB and 3 replicas
+    And Attach volume 0 to node 1
+    And Wait for volume 0 healthy
+    And Update volume 0 replica count to 2
+    And Delete volume 0 replica on volume node
+
+    When Keep writing data to volume 0
+    And Disconnect volume 0 node network for 100 seconds without waiting for completion
+
+    Then Wait for volume 0 detached
+    And Wait for volume 0 faulted
+
+Node Disconnect And Keep Data Writing
+    Given Set setting auto-salvage to false
+
+    When Create volume 0 with 10 GB and 3 replicas
+    And Attach volume 0 to node 1
+    And Wait for volume 0 healthy
+
+    When Keep writing data to volume 0
+    And Disconnect volume 0 node network for 100 seconds without waiting for completion
```

Comment:
The steps looking good but when a node is disconnected from a pod, it is impossible to delete that pod to restore the connection because the traffic to the node is not available at that time.